boot/freeldr/freeldr/arch/drivers/hwide.c

```diff
@@ -461,11 +461,48 @@ AtapiCapacityDetect(
 
         *SectorSize = (AtapiCapacity[4] << 24) | (AtapiCapacity[5] << 16) |
                       (AtapiCapacity[6] << 8) | AtapiCapacity[7];
+
+        /* Original value isn't needed because we use READ command instead of READ CD */
+        if (*SectorSize != 0)
```

Comment:
Maybe at least we can check for the standard returned size (should be size == 2352, the "RAW sector size") to at least somewhat validate the fact the ATAPI/IDE interface was really reading a CD there.

Reply:
Emulators returns the 2048 value here...